File: Count number of hops - GFG/count-number-of-hops.java
//{ Driver Code Starts
//Initial Template for Java

import java.util.*;
import java.io.*;
import java.lang.*;

class GFG implements Runnable {
    public void run() {
        try {
            BufferedReader in;
            PrintWriter out;
            in = new BufferedReader(new InputStreamReader(System.in));
            out = new PrintWriter(System.out);
            int t = Integer.parseInt(in.readLine());

            while (t-- > 0) {
                int n = Integer.parseInt(in.readLine().trim());

                Solution g = new Solution();
                out.println(g.countWays(n));
            }
            out.close();
        } catch (Exception e) {
            ;
        }
    }

    public static void main(String args[]) throws IOException {
        new Thread(null, new GFG(), "whatever", 1 << 26).start();
    }
}
// } Driver Code Ends


//User function Template for Java


class Solution
{
    //Function to count the number of ways in which frog can reach the top.
    static long countWays(int n)
    {
        // add your code here
if(n==1) return 1L;
        
        int[] dp=new int[n+1];
        dp[0]=1;
        
        int mod=1000000007;
        
        for(int i=1;i<=n;i++){
            int one=dp[i-1];
            int two=0,three=0;
            if(i>=2) two=dp[i-2];
            if(i>=3) three=dp[i-3];
            dp[i]=((one%mod+two%mod)%mod+three%mod)%mod;
        }
        
        return dp[n];
    }
    
}

File: Count number of hops - GFG/count-number-of-hops.py
#User function Template for python3

class Solution:
    #Function to count the number of ways in which frog can reach the top.
    def countWays(self,n):
        
        # code here
        if n <= 1:
            return 1

        MOD = 10**9 + 7
        # Initialize a list to store the number of ways to reach each step
        ways = [0] * (n + 1)
        ways[0] = 1  # There's one way to stay at step 0
        ways[1] = 1  # There's one way to reach step 1

        for i in range(2, n + 1):
            # Calculate the number of ways to reach step i by summing up
            # the ways to reach the previous 3 steps
            ways[i] = (ways[i - 1] + ways[i - 2] + ways[i - 3]) % MOD

        return ways[n] % MOD

#{ 
 # Driver Code Starts
#Initial Template for Python 3
import atexit
import io
import sys
sys.setrecursionlimit(10**6)
# Contributed by : Nagendra Jha

if __name__ == '__main__':
    test_cases = int(input())
    for cases in range(test_cases):
        m = int(input())
        ob=Solution()
        print(ob.countWays(m))
# } Driver Code Ends